tests/unit/steps/test_steps_plan.py

```diff
@@ -162,14 +162,74 @@ async def test_generate_plan(apps, model):
     assert upgrade_plan == expected_plan
 
 
-@pytest.mark.asyncio
-async def test_generate_plan_raise_NoTargetError(mocker):
-    exp_error_msg = "Cannot find target to upgrade."
-    analysis_result = MagicMock(spec=Analysis)
-    analysis_result.current_cloud_os_release.next_release = None
-    # not possible to determine target
+@pytest.mark.parametrize(
+    "current_os_release, current_series, next_release",
+    [
+        (OpenStackRelease("victoria"), "focal", "wallaby"),
+        (OpenStackRelease("yoga"), "jammy", "zed"),
```

Comment:
This test should fail IMO. If a user manually upgrade the series to jammy and run COU I would fell more comfortable not letting go ahead and upgrade to zed.

Reply:
As we are including the range limit, I revised the tests